tests/workflow_tests/test_events.py

```diff
@@ -63,17 +63,25 @@ def test_sorted_events(self):
         self.assertListEqual(events.items, sorted_events)
 
     def test_filtered_events(self):
-        deployment_id = self.deployment_id
-        # create another deployment to test correct filtering
-        self._create_deployment()
-        filters = {'deployment_id': deployment_id}
+        # create multiple deployments
+        deployment_ids = []
+        for i in range(3):
+            deployment_ids.append(self._create_deployment())
+
+        # filter a subset of the deployments
+        expected_deployment_ids = deployment_ids[:2]
+        filters = {'deployment_id': expected_deployment_ids}
         events = self.client.events.list(**filters)
+
         self.assertGreater(len(events), 0, 'No events')
-        for event in events:
-            self.assertEqual(event['context']['deployment_id'],
-                             deployment_id,
-                             'Expected only events related to'
-                             ' deployment id {0}'.format(deployment_id))
+        deployments_with_events = \
+            set([event['context']['deployment_id'] for event in events])
+        self.assertEquals(sorted(deployments_with_events),
+                          sorted(expected_deployment_ids),
+                          'Expected only events of deployment ids {0}, '
```

Comment:
error message assumes there might be additional deployment ids, though some might be missing out rather than have additional ones
i'd change the phrasing